src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrVersionFileMerger.cs

```diff
@@ -91,23 +91,27 @@ public async Task MergeJsonAsync(
             : targetRepoPreviousJson;
         var vmrCurrentJson = await GetJsonFromGit(vmr, vmrJsonPath, vmrCurrentRef, allowMissingFiles);
 
-        var targetRepoChanges = SimpleConfigJson.Parse(targetRepoPreviousJson).GetDiff(SimpleConfigJson.Parse(targetRepoCurrentJson));
-        var vmrChanges = SimpleConfigJson.Parse(vmrPreviousJson).GetDiff(SimpleConfigJson.Parse(vmrCurrentJson));
+        if (!(allowMissingFiles && DeleteJsonFileIfRequiredAsync(
+                targetRepoPreviousJson,
+                targetRepoCurrentJson,
+                vmrPreviousJson,
+                vmrCurrentJson,
+                targetRepo.Path / jsonRelativePath)))
```

Comment:
I'd negate the inside to `!allowMissingFiles || !DeleteJsonFileIfRequiredAsync(..)` so that if `allowMissingFiles` is `false`, it does not evaluate `DeleteJsonFileIfRequiredAsync`.